### ISSUE TITLE
IDE0028 has changed between .NET 8 Preview 7 and RC1

### ISSUE THREAD

**Issue by bording** [dotnet/roslyn#69940]
**Version Used**: .NET 8.0.100-rc.1.23455.8 SDK

**Steps to Reproduce**:

1. Create a new project with `dotnet new classlib`
2. Add `<EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>` to the csproj
3. Add an .editorconfig file with `dotnet_diagnostic.IDE0028.severity = error`
4. [Add code the triggers the analyzer](https://github.com/bording/ide0028test/blob/main/ide0028test/Test.cs)

I have provided sample code to reproduce this [here](https://github.com/bording/ide0028test).

**Diagnostic Id**: This is related to analyzers reporting on `IDE0028`.

**Expected Behavior**:

The documented behavior for `IDE0028` is https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0028

There is nothing about the code that should be triggering `IDE0028` as documented, so I expect it to compile without an error.

**Actual Behavior**:

The code when compiled with `dotnet build` fails with 2 errors:

```
C:\Code\ide0028test\ide0028test\Test.cs(14,31): error IDE0028: Collection initialization can be simplified (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0028) [C:\Code\ide0028test\ide0028test\ide0028test.csproj]
C:\Code\ide0028test\ide0028test\Test.cs(7,24): error IDE0028: Collection initialization can be simplified (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0028) [C:\Code\ide0028test\ide0028test\ide0028test.csproj]
```

The URL does not seem to match the error.

This code compiles fine without an error using the 8.0.100-preview.7.23376.3 SDK or earlier. If I change the TFM to `net7.0`, I can also compile it without error using the 7.0.401 SDK. 

`IDE0028` as documented has been around for a while and we've been enforcing it. It seems that some new rule as been added that is also reporting as `IDE0028` and has different requirements. This should be a new rule and not reusing `IDE0028`.

I'm also only seeing this error when compiling from the command line via `dotnet build`. It is not causing an error in Visual Studio Version 17.8.0 Preview 1.0.

**Comment by CyrusNajmabadi** [comment]
There are new ways that collections can be simplified. This analyzer/fixer has picked those up. This is a normal thing we do as the language improves or the patterns that we detect are expanded.

**Comment by bording** [comment]
@CyrusNajmabadi This seems like a really bad idea to just have an existing rule start flagging completely unrelated things. Have to say that I completely disagree with this.

Where's the documentation that explains what these new things are? This isn't even surfacing in Visual Studio itself, so I had to guess at what it was even wanting me to do because the documentation the error linked to doesn't even say anything about what the analyzer is looking for.

**Comment by CyrusNajmabadi** [comment]
> This seems like a really bad idea to just have an existing rule start flagging completely unrelated things. 

It's not unrelated. It's just more cases that are found and fixed. Which is how our analyzers and fixers have always worked. 

> This isn't even surfacing in Visual Studio itself,

It will surface in the next public preview of VS. This happens though when taking preview releases of our analyzers as the previews for each are on different cadences.  The final releases do sync up though. 

> Also, given that this is a code style analyzer, I'm not sure I understand how it's justified for this rule to "require" specifying an initial capacity. How is that code style and not a code analysis/performance thing instead?

That's not the fix. It's simply that if you provide a capacity that we do not report the diagnostic anymore. So, in essence, you suppressed the style analyzer by indicating that performance matters here with a capacity.

Thanks!

**Comment by bording** [comment]
> That's not the fix. It's simply that if you provide a capacity that we do not report the diagnostic anymore. So, in essence, you suppressed the style analyzer by indicating that performance matters here with a capacity.

That's good to know. Are there plans to update the documentation to match these new analyzer changes? If so, if there an issue I can follow related to it?

My frustration here is really around there not being any way for me to know what the analyzer wants me to do. If the documentation already had the relevant information, or if there were some release notes, or really any source of information I could have looked at regarding this, then I wouldn't have felt the need to open the issue in the first place.

**Comment by CyrusNajmabadi** [comment]
> My frustration here is really around there not being any way for me to know what the analyzer wants me to do. If the documentation already had the relevant information, or if there were some release notes, or really any source of information I could have looked at regarding this, then I wouldn't have felt the need to open the issue in the first place.

Different tools/system get updates on different cadences.  We often do the analyzer/fixer work first, and that makes it out immediately, with the rest of our ecosystem components updating in the interim between then and RTM.  

Thanks! :)
